src/NetscriptFunctions/Singularity.ts

```diff
@@ -687,6 +689,41 @@ export function NetscriptSingularity(): InternalAPI<ISingularity> {
         .filter((_position) => Companies[companyName].hasPosition(_position[0]))
         .map((_position) => _position[1]["name"]);
     },
+    getCompanyPositionInfo: (ctx) => (_companyName, _positionName) => {
+      helpers.checkSingularityAccess(ctx);
+      const companyName = helpers.string(ctx, "companyName", _companyName);
+      const positionName = helpers.string(ctx, "positionName", _positionName);
+
+      // Make sure its a valid company
+      if (companyName == null || companyName === "" || !Companies[companyName]) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid company: '${companyName}'`);
+      }
+
+      // Make sure its a valid position
+      if (positionName == null || positionName === "" || !CompanyPositions[positionName]) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid position: '${positionName}'`);
+      }
```

Comment:
This doesn't correctly narrow the type for positionName from `string` to `CompanyPosName`.
For typechecking something that has an enum defined, currently the solution is using `checkEnum`, see NetscriptFunctions/Infiltration.ts function `calculateInfiltrationData` for an example. After throwing an error if checkEnum returned false, the type is narrowed to a member of the provided enum.

There's another PR in progress that will replace checkEnum with something else, but the basic idea will remain the same.